independent-projects/resteasy-reactive/common/runtime/src/main/java/org/jboss/resteasy/reactive/common/headers/CookieParser.java

```diff
@@ -0,0 +1,185 @@
+package org.jboss.resteasy.reactive.common.headers;
+
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+import java.util.Locale;
+import java.util.NoSuchElementException;
+import java.util.StringTokenizer;
+import javax.ws.rs.core.NewCookie;
+
+public class CookieParser {
+
+    private static final String SET_COOKIE = "set-cookie:";
+    private static final String SET_COOKIE2 = "set-cookie2:";
+
+    private static final String[] COOKIE_DATE_FORMATS = {
+            "EEE',' dd-MMM-yyyy HH:mm:ss 'GMT'",
+            "EEE',' dd MMM yyyy HH:mm:ss 'GMT'",
+            "EEE MMM dd yyyy HH:mm:ss 'GMT'Z",
+            "EEE',' dd-MMM-yy HH:mm:ss 'GMT'",
+            "EEE',' dd MMM yy HH:mm:ss 'GMT'",
+            "EEE MMM dd yy HH:mm:ss 'GMT'Z"
+    };
+
+    public static NewCookie parse(String header) {
+        if (startsWithIgnoreCase(header, SET_COOKIE2)) {
+            header = header.substring(SET_COOKIE2.length());
+        } else if (startsWithIgnoreCase(header, SET_COOKIE)) {
+            header = header.substring(SET_COOKIE.length());
+        }
+
+        return parseInternal(header);
+    }
+
+    private static boolean startsWithIgnoreCase(String s, String start) {
+        if (s == null || start == null)
+            return false;
+
+        if (s.length() >= start.length() &&
+                start.equalsIgnoreCase(s.substring(0, start.length()))) {
```

Comment:
Can be simplified to: `s.regionMatches(true, 0, start, 0, start.length())`

Reply:
This is coming from the JDK so if we go this route, I would keep the logic as equivalent as possible.

Anyway, let's decide on a direction first before entering into the details.